### ISSUE TITLE
Request: config option to respect MTA's "From:" header handling

### ISSUE THREAD

**Issue by scruloose** [mvo5/unattended-upgrades#248]
I have recently started using U-U on several machines, in conjunction with `msmstp-mta` and `bsd-mailx` to capture "local" mail and send it via a smarthost.

I have msmtp configured to automatically generate the "From" header by appending `@<hostname>.<domain>.<tld>` to the username of the account sending the mail, and I have verified that it works correctly when sending test e-mail, eg: \
`echo "This is the message text" | mail -s "test message from $HOSTNAME" root`

Experimenting with the `Unattended-Upgrade::Sender` config option, I have not found any way to instruct U-U to *not* set the "From" header. The default, as mentioned in README.md, is "root". Which is fine, but explicitly setting the "From:" header overrides `msmtp`'s `auto-from` feature. In conjunction with the fact that U-U's hard-coded subject line is long enough to truncate to "unattended-upgrade result f..." on my mail client, this means I can't actually see the key information of "which host(s) have what U-U result" by skimming my inbox, and have to open each email one by one to find that out.

I tried explicitly setting `Unattended-Upgrade::Sender ""` in the hopes that I could *unset* the default, but it made no difference.

As a workaround, I can hard-code the hostname in `50unattended-upgrades`, but it would be preferable to have the option of keeping the config file portable and letting my msmtp config handle this.

Would it be fairly straightforward to check for a special value (like "None", or "%mta-default" or whatever) for `Unattended-Upgrade::Sender` and not set the header at all if the config option is set to that value?

**Comment by rbalint** [comment]
If you would like to keep the `50unattended-upgrades` file portable you can also place a per-host `51unattended-upgrades-sender` overriding the `Sender` option.

**Comment by scruloose** [comment]
True, but that doesn't solve the real problem—having to hard-code the hostname *somewhere* in the config for unattended-upgrades (reproducing information that's readily available from the OS *and* already used as a variable in U-U itself) for no purpose except to reproduce functionality that my MTA [already handles](https://git.marlam.de/gitweb/?p=msmtp.git;a=commit;h=f498ac0ac0777c33825476e43e2b975e5714235b) much more elegantly. It would be preferable to have the option to tell U-U to step aside and let msmtp do its job, even if that's not the default.

**Comment by Gunni** [comment]
I just hit this issue myself, disappointing to see a ticket from 2020 still unfixed...
